soc/silabs/silabs_s2/xg23/Kconfig.soc

```diff
@@ -13,12 +14,24 @@ config SOC_SERIES_EFR32ZG23
 	help
 	  Silicon Labs EFR32ZG23 Series MCU
 
+config SOC_SERIES_EFM32PG23
+	bool
+	select SOC_SILABS_XG23
+	help
+	  Silicon Labs EFM32PG23 Series MCU
+
 config SOC_PART_NUMBER_EFR32ZG23B020F512IM48
 	bool
 	select SOC_SERIES_EFR32ZG23
 
+config SOC_EFM32PG23B310F512IM48
```

Comment:
The other parts are named `SOC_PART_NUMBER_*`. Any reason to not follow this pattern?

Reply:
soc part number is wrong, no where is that documented in the porting guide

Reply:
see the previous discussion in this PR at https://github.com/zephyrproject-rtos/zephyr/pull/93285#discussion_r2230559319